code/datums/hud/meson_hud.dm

```diff
@@ -45,19 +45,22 @@ var/list/meson_images = list()
 		register_event(/event/moved, src, nameof(src::update_meson_image()))
 
 /atom/movable/Destroy()
+	if(meson_image)
+		is_on_mesons = FALSE
+		update_meson_image()
+		unregister_event(/event/moved, src, nameof(src::update_meson_image()))
+	..()
+
+/atom/movable/proc/set_on_mesons(var/set_on = 0)
+	is_on_mesons = set_on
+	update_meson_image()
```

Comment:
Wouldn't you want to add/remove the new event in here too?

Reply:
ah yes, moved it to a better place now